lib/runtime-c-api/src/trampoline.rs

```diff
@@ -61,7 +61,7 @@ pub unsafe extern "C" fn wasmer_trampoline_buffer_builder_build(
 #[allow(clippy::cast_ptr_alignment)]
 pub unsafe extern "C" fn wasmer_trampoline_buffer_destroy(buffer: *mut wasmer_trampoline_buffer_t) {
     if !buffer.is_null() {
-        Box::from_raw(buffer);
+        Box::from_raw(buffer as *mut TrampolineBuffer);
```

Comment:
We usually use the following syntax:

```rs
let _: Box<TrampolineBuffer> = Box::from_raw(buffer);
```

Technically, it's the same though.

However, what surprises me is: What type `Box::from_raw` returns in this case? `Box<wasmer_trampoline_buffer>`? In this case, it's size is 1, and so, yes, it leaks data.